rct229/rulesets/ashrae9012019/section11/section11rule12.py

```diff
@@ -0,0 +1,156 @@
+    " 2. Facilities that provide 60% of their service water heating from site-solar energy or siterecovered energy or from other sources.) Recommend manual review to determine if project complies."
+)
+
+
+class Section11Rule12(RuleDefinitionListIndexedBase):
+    """Rule 12 of ASHRAE 90.1-2019 Appendix G Section 11 (Service Water Heating)"""
+
+    def __init__(self):
+        super(Section11Rule12, self).__init__(
+            rmds_used=produce_ruleset_model_description(
+                USER=False, BASELINE_0=True, PROPOSED=True
+            ),
+            each_rule=Section11Rule12.RMDRule(),
+            index_rmd=BASELINE_0,
+            id="11-12",
+            description="For large, 24-hour-per-day facilities that meet the prescriptive criteria for use of condenser heat recovery systems described in Section 6.5.6.2, a system meeting the requirements of that section shall be included in the baseline building design regardless of the exceptions to Section 6.5.6.2.",
+            ruleset_section_title="Service Water Heating",
+            standard_section="Table G3.1 #11, baseline column, d + exception",
+            is_primary_rule=False,
+            list_path="ruleset_model_descriptions[0]",
+            data_items={"is_leap_year_b": (BASELINE_0, "calendar/is_leap_year")},
+        )
+
+    class RMDRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section11Rule12.RMDRule, self).__init__(
+                rmds_used=produce_ruleset_model_description(
+                    USER=False, BASELINE_0=True, PROPOSED=True
+                ),
+                each_rule=Section11Rule12.RMDRule.BuildingRule(),
+                index_rmd=BASELINE_0,
+                list_path="$.buildings[*]",
+            )
+
+        def create_data(self, context, data):
+            rmd_b = context.BASELINE_0
+            rmd_p = context.PROPOSED
+
+            return {"rmd_b": rmd_b, "rmd_p": rmd_p}
+
+        class BuildingRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section11Rule12.RMDRule.BuildingRule, self).__init__(
+                    rmds_used=produce_ruleset_model_description(
+                        USER=False, BASELINE_0=True, PROPOSED=True
+                    ),
+                    each_rule=Section11Rule12.RMDRule.BuildingRule.BuildingSegmentRule(),
+                    index_rmd=BASELINE_0,
+                    list_path="$.building_segments[*]",
+                )
+
+            def create_data(self, context, data):
+                building_b = context.BASELINE_0
+                rmd_b = data["rmd_b"]
+                is_leap_year_b = data["is_leap_year_b"]
+
+                bldg_open_sch_id_b = getattr_(
+                    building_b, "buildings", "building_open_schedule"
+                )
+
+                bldg_open_sch_b = find_exactly_one_schedule(rmd_b, bldg_open_sch_id_b)[
+                    "hourly_values"
+                ]
```

Comment:
Following the above `create_data` suggestion, you can use `find_exactly_one_with_field_value("$.schedules_b[*]", "id", bldg_open_sch_id_b, data)` to get the bldg_open_sch_b.

Reply:
Addressed.